system/lib/wasmfs/syscalls.cpp

```diff
@@ -513,4 +520,85 @@ __wasi_errno_t __wasi_fd_fdstat_get(__wasi_fd_t fd, __wasi_fdstat_t* stat) {
   }
   return __WASI_ERRNO_SUCCESS;
 }
+
+// This enum specifies whether rmdir or unlink is being performed.
+enum class UnlinkMode { Rmdir, Unlink };
+
+static long __unlink(char* path, UnlinkMode unlinkMode) {
```

Comment:
The `__` prefix is reserved for system use, and since this isn't a system call itself, it would be better to use a non-reserved name.
```suggestion
static long doUnlink(char* path, UnlinkMode unlinkMode) {
```